### DISCUSSION TITLE
Make Multithreading as Default in Android

### DISCUSSION THREAD

**Discussion by Shirou27** [ppy/osu#23080] [Ideas] [answered]
Most Android phones run games better at Multithreading than single thread

Base on my knowledge, it is rare for an Android phone to run games better at single thread than Multi thread.

New Android players might thought their phone can't handle better performance or the game is very laggy since they don't know that they should change or try to change their threading mode to Multithreading inside settings of osu! Lazer.

I know new renderer for Android will come soon but at the same time, you should also set the default threading mode to Multithreading too.

**Comment by bdach** [comment, marked as answer]
Single thread is used on mobile intentionally to save battery and avoid devices overheating. If you're sure your device can handle it, then you can switch yourself, that's why it's a switch.

**Reply by peppy** [reply]
In addition, android exposes an API similar to iOS which should allow for frames to be timed accurately and pushed to the GPU to get very low latency. Running extra threads to brute force frame timings only adds heat/battery overheads.
